v1/logging/config.go

```diff
@@ -0,0 +1,140 @@
+package logging
+
+import (
+	"log/syslog"
+
+	validation "github.com/go-ozzo/ozzo-validation/v4"
+	"github.com/go-ozzo/ozzo-validation/v4/is"
+	"gopkg.in/inconshreveable/log15.v2"
+)
+
+var (
+	defaultLogLevel  = "error"
+	defaultLogFormat = "plain"
+)
+
+// LogDestinationConfig represents a logging reporter destination.
+type LogDestinationConfig struct {
+	// Type represents the destination type (file|console|syslog).
+	Type string `yaml:"type"`
+
+	// Destination represents the log destination depending on the type:
+	// - For type "file", is must be a filesystem path
+	// - For "console", it is ignored
+	// - For type "syslog", it can be either empty (local syslog) or a net.Dial format string  for remote syslog
+	Destination string `yaml:"destination"`
+
+	// Level represents the highest message severity level to report (crit..debug).
+	Level string `yaml:"level"`
+
+	// Format represents the format to apply to logged messages (plain|json).
+	Format string `yaml:"format"`
+}
+
+func (c *LogDestinationConfig) logFormat() log15.Format {
+	switch c.Format {
+	case "json":
+		return log15.JsonFormat()
+
+	default:
+		return log15.LogfmtFormat()
+	}
+}
+
+func (c *LogDestinationConfig) validate() error {
+	if c.Level == "" {
+		c.Level = defaultLogLevel
+	}
+
+	if c.Format == "" {
+		c.Format = defaultLogFormat
+	}
+
+	return validation.ValidateStruct(c,
+		validation.Field(&c.Type,
+			validation.Required,
+			validation.In(
+				"file",
+				"console",
+				"syslog",
+			)),
+
+		validation.Field(&c.Destination,
+			validation.When(c.Type == "file", validation.Required),
+			validation.When(c.Type == "syslog" && c.Destination != "", is.DialString)),
+
+		validation.Field(&c.Level,
+			validation.By(func(v interface{}) error {
+				_, err := log15.LvlFromString(v.(string))
+				return err
+			})),
+
+		validation.Field(&c.Format,
+			validation.In(
+				"plain",
+				"json",
+			)),
+	)
+}
+
+// Config represents a logging reporter configuration.
+type Config struct {
+	Destinations []*LogDestinationConfig `yaml:"destination"`
```

Comment:
I think it's not necessary to put a slice of pointer here ;)

Reply:
I think it is, since the `LogDestinationConfig.validate()` method can modify the receiver if it needs to set a default value for one of the fields: https://github.com/exoscale/go-reporter/pull/15/files#diff-c60c6317714d274c9e1053b6b1ab40bbR45-R51

Reply:
👍